### PR TITLE
docs: update docstrings for dataframe.schema

### PR DESCRIPTION
**Pull request by mfonekpo**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [ ] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [x] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Related issues 

- Related issue # 
- Closes #

## Checklist

- [ ] Code follows style guide (ruff)
- [ ] Tests added 
- [x] Documented the changes

### CODE REVIEW

File: narwhals/dataframe.py [hunk 1]
```diff
@@ -568,16 +572,28 @@ def schema(self) -> dict[str, DType]:
...
+            ...     df = nw.from_native(df_any)
+            ...     return df.schema
+
+            You can pass either pandas or Polars to `func`:
+
+            >>> df_pd_schema = func(df_pd)
+            >>> df_pd_schema = func(df_pl)
+
+            Both pandas and Polars returns the same output:
             OrderedDict({'foo': Int64, 'bar': Float64, 'ham': String})
```
Comment (MarcoGorelli) [inline narwhals/dataframe.py:597]: show the output for each, so it gets tested in CI (see other docstrings for examples)

Comment (mfonekpo) [reply narwhals/dataframe.py:597]: Ok, on it

File: narwhals/dataframe.py [hunk 2]
```diff
@@ -568,16 +572,33 @@ def schema(self) -> dict[str, DType]:
...
+            >>> df_pd_schema = func(df_pd)
+            >>> df_pd_schema
+            {'foo': Int64, 'bar': Float64, 'ham': String}
+
+            >>> df_pl_schema = func(df_pl)
+            >>> df_pl_schema
+            {'foo': Int64, 'bar': Float64, 'ham': String}
+
+            Both pandas and Polars returns the same output:
             OrderedDict({'foo': Int64, 'bar': Float64, 'ham': String})
```
Comment (MarcoGorelli) [inline narwhals/dataframe.py:602]: let's remove this

Comment (mfonekpo) [reply narwhals/dataframe.py]: Done @MarcoGorelli

**Pull request merged**